lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Trento.DeregistrationProcessManager do
+  @moduledoc """
+    DeregistrationProcessManager is a Commanded ProcessManager, it's the responsible
+    for the deregistration procedure for both hosts and clusters.
+
+    This represent a transaction to ensure that the procedure of deregistering hosts and cluster
+    follows a certain path and satisfies some requisities.
+
+    For more information see https://hexdocs.pm/commanded/process-managers.html
+  """
+
+  @required_fields :all
+
+  use Trento.Type
+
+  use Commanded.ProcessManagers.ProcessManager,
+    application: Trento.Commanded,
+    name: "deregistration_process_manager"
+
+  deftype do
+    field :host_id, Ecto.UUID
+  end
+
+  alias Trento.DeregistrationProcessManager
```

Comment:
It is not a bit strange to self-alias the same module?
In other parts of the code we have used `__MODULE__`